waspc/src/Wasp/CompileOptions.hs

```diff
@@ -16,5 +17,6 @@ data CompileOptions = CompileOptions
     -- We give the compiler the ability to send messages. The code that
     -- invokes the compiler (such as the CLI) can then implement a way
     -- to display these messages.
-    sendMessage :: SendMessage
+    sendMessage :: SendMessage,
+    warningsFilter :: [GeneratorWarning] -> [GeneratorWarning]
```

Comment:
This is where this field is defined, so this is also where we should provide rich comment explaining its purpose and intention.

Reply:
Sounds good! I can add a comment and can update the name 👍🏻 Thanks!